internal/process/update/update_runtime_step.go

```diff
@@ -0,0 +1,91 @@
+package update
+
+import (
+	"context"
+	"time"
+
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type UpdateRuntimeStep struct {
+	operationManager *process.OperationManager
+	k8sClient        client.Client
+}
+
+func NewUpdateRuntimeStep(os storage.Operations, k8sClient client.Client) *UpdateRuntimeStep {
+	return &UpdateRuntimeStep{
+		operationManager: process.NewOperationManager(os),
+		k8sClient:        k8sClient,
+	}
+}
+
+func (s *UpdateRuntimeStep) Name() string {
+	return "Update_Runtime_Resource"
+}
+
+func (s *UpdateRuntimeStep) Run(operation internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	// Check if the runtime exists
+
+	var runtime = imv1.Runtime{}
+	err := s.k8sClient.Get(context.Background(), client.ObjectKey{Name: operation.GetRuntimeResourceName(), Namespace: operation.GetRuntimeResourceNamespace()}, &runtime)
+	if errors.IsNotFound(err) {
+		// todo: after the switch to KIM, this should throw an error
+		log.Infof("Runtime not found, skipping")
+		return operation, 0, nil
+	}
+
+	// Update the runtime
```

Comment:
We have created two auxiliary functions to handle assignment depending on nil/zero value. 
```
runtime.Spec.Shoot.Provider.Workers[0].Machine.Type = DefaultIfParamNotSet(operation.UpdatingParameters.MachineType, operation.UpdatingParameters.MachineType)
```